src/TreeView/TreeView.tsx

```diff
@@ -78,41 +80,62 @@ export type TreeViewItemProps = {
   children: React.ReactNode
   current?: boolean
   defaultExpanded?: boolean
+  expanded?: boolean
+  onExpandedChange?: (expanded: boolean) => void
   onSelect?: (event: React.MouseEvent<HTMLElement> | KeyboardEvent) => void
-  onToggle?: (isExpanded: boolean) => void
 }
 
 const Item: React.FC<TreeViewItemProps> = ({
-  current: isCurrent = false,
+  current: isCurrentItem = false,
   defaultExpanded = false,
+  expanded,
+  onExpandedChange,
   onSelect,
-  onToggle,
   children
 }) => {
   const {setActiveDescendant} = React.useContext(RootContext)
   const itemId = useSSRSafeId()
   const labelId = useSSRSafeId()
   const itemRef = React.useRef<HTMLLIElement>(null)
-  const {level} = React.useContext(ItemContext)
-  const [isExpanded, setIsExpanded] = React.useState(defaultExpanded)
+  const [isExpanded, setIsExpanded] = useControllableState({
+    name: itemId,
+    defaultValue: defaultExpanded,
+    value: expanded,
+    onChange: onExpandedChange
+  })
+  const [containsCurrentItem, setContainsCurrentItem] = React.useState(false)
+  const {level, setContainsCurrentItem: setParentContainsCurrent} = React.useContext(ItemContext)
   const {hasSubTree, subTree, childrenWithoutSubTree} = useSubTree(children)
 
+  // Create stable function references
+  const setIsExpandedRef = React.useRef(setIsExpanded)
+  const setParentContainsCurrentRef = React.useRef(setParentContainsCurrent)
+
+  React.useLayoutEffect(() => {
+    setIsExpandedRef.current = setIsExpanded
+  })
+
+  React.useLayoutEffect(() => {
+    setParentContainsCurrentRef.current = setParentContainsCurrent
```

Comment:
With `setContainsCurrentItem` coming from `React.useState()`, we might be able to rely on that being stable instead of having to create a stable reference.